packages/core/src/lib/select/select-menu.svelte

```diff
@@ -0,0 +1,72 @@
+<svelte:options
+  immutable
+  accessors
+/>
+
+<script lang="ts">
+import cx from 'classnames';
+import { Icon } from '$lib';
+
+export let open = false;
+export let element: HTMLUListElement;
+export let button: { text: string; icon?: string } | undefined = undefined;
+export let heading = '';
+</script>
+
+<div
+  class={cx(
+    'border-gray-9 absolute left-0 right-0 top-full z-40 mt-1 w-full overflow-hidden border bg-white drop-shadow-md',
+    {
+      invisible: !open,
+    }
+  )}
+>
+  {#if heading}
+    <span class="text-default flex flex-wrap py-1 pl-2 text-xs">
+      {heading}
+    </span>
+  {/if}
+  <ul
+    role="menu"
+    class="select-menu flex max-h-36 flex-col overflow-y-auto"
+    bind:this={element}
+    on:mouseleave
+  >
+    <slot />
+  </ul>
+  {#if button !== undefined}
+    <button
```

Comment:
What's this button for? I think we'll need an explicit `type=button` to prevent it becoming an implicit submit button if it's used inside a `<form>`

Reply:
This is a bit of a holdover from the previous `select` but it is also shown in this layout here:

<img width="218" alt="Screenshot 2023-08-24 at 2 53 34 PM" src="https://github.com/viamrobotics/prime/assets/1930371/fe46c1d1-d70e-42cf-b591-2192fee2b5c3">

I don't love this setup really, would probably prefer to do it with slots. Actually maybe I will just do that.

Reply:
Hm actually can't use slots all the way down because of rendering svelte fragments inside of conditionals, I will continue to explore some options but until then I can make sure we get the `type` on there.